src/main/java/net/glowstone/entity/GlowLivingEntity.java

```diff
@@ -997,8 +998,22 @@ public TaskManager getTaskManager() {
     public boolean entityInteract(GlowPlayer player, InteractEntityMessage message) {
         super.entityInteract(player, message);
 
+        if (message.getAction() != Action.INTERACT.ordinal()) {
```

Comment:
This this check should be done inside the `isLeashed() && [...]` check, because you don't want to return `false` for every other action type that other entities can implement.